Ft update profile

chrill-christella opened the pull request:
### what does this PR do?

update the profile of the user

### Description of the task to be completed

- generate the user profile whenever a user sign up or login 
- create user profile controller and route to update the profile of a buyer or a sender
- add necessary validations

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

chrill-christella committed:
update user profile:
user profile controller
user profile service
user profile route
[finish]

sostene12 committed:
ft(roles-permission): roles and permissions

- implement roles and permission relationship
- implement create role and delete role
- implement create and delete permission
- implement assignrole and permission to the user

[WIP ft-roles-permission ]

chrill-christella committed:
--amend

Shumba250 committed:
ft-register-vendor:
the administrator should be able to register a new vendor
an email containing a rondom password should be sent to the provided email
the vendor should be able to signin using the provided email and password
appropriate aeeror messages are displayed on unsuccessful attempts

chrill-christella committed:
ft-update-profile:
create a profile service
 create profile controller as well as route
 swagger doc
testing the endpoint
[finish]

sostene12 reviewed (changes requested):
please consider the conflicts you are having, it would help if you solve them first and then rebase the base branch (develop)

sostene12 committed:
ft(roles-permission): roles-permission

-implement internationalization
-implement swgger doc of the endpoint
-implement enpoints test

[finished ft-roles-permissions ]

sostene12 committed:
ft-roles-permissions
- implement relationships

patsicko reviewed (changes requested):
- We are normalizing user model by creating user profiles with respect to their respective roles.
I can suggest you to rebase ft: roles permissions #20 to have the latest changes. It will help.
- CI.yml has been changed to run a script that undo all migrations after each test. The same changes have been applied to the test script in package.json.  Hopefully this will solve the issues of your tests that are not passing.

sostene12 commented on src/api-docs/buyer.docs.js:
you are sending the id of user, hence you are finding by email from req.body

```
static async getProfile (req, res, next) {
  try {
    const { userId } = req.params;
    const {dataValues} = await user.findOne({where:{email:req.body.email}});
    const profile = await profiles.findUserProfile(userId);
    const {...others}= dataValues;
    if (profile) {
      res.json({ message: req.t('Found'), data: {others,profile} });
    } else {

      const Message = req.t('notfound');
      res.json({ message: Message });
    }
  } catch (error) {
    res.status(500).json({ error: error.message });
    next(error);
  }
```

sostene12 commented on src/routes/buyer.route.js:
it can be good , if before updating you must login first

chrill-christella replied:
resolved this issue too

sostene12 commented:
also, solve conflicts, and apply the comments on the vendor aswell
